### PR TITLE
[cloud_functions] update Dart API to replace call with getHttpsCallable

### PR DESCRIPTION
**Pull request by collinjackson**
Updates cloud_functions API to more closely match native SDK API

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [X] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [ ] No, this is *not* a breaking change.

### CODE REVIEW

File: packages/cloud_functions/lib/src/https_callable.dart [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  final String _functionName;
+
+  Future<HttpsCallableResult> call([dynamic parameters]) async {
+    try {
+      final MethodChannel channel = CloudFunctions.channel;
+      final dynamic response =
+          // TODO(amirh): remove this on when the invokeMethod update makes it to stable Flutter.
+          // https://github.com/flutter/flutter/issues/26431
+          // ignore: strong_mode_implicit_dynamic_method
+          await channel.invokeMethod('CloudFunctions#call', <String, dynamic>{
```
Comment (bparrishMines) [inline packages/cloud_functions/lib/src/https_callable.dart]: nit: You could use the new `invokeMethod<T>` here and remove the comment above.

File: packages/cloud_functions/lib/src/https_callable.dart [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        'functionName': _functionName,
+        'parameters': parameters,
+      });
+      return HttpsCallableResult._(response);
+    } on PlatformException catch (e) {
+      if (e.code == 'functionsError') {
+        final String code = e.details['code'];
+        final String message = e.details['message'];
+        final dynamic details = e.details['details'];
+        print('throwing firebase functions exception');
```
Comment (bparrishMines) [inline packages/cloud_functions/lib/src/https_callable.dart]: nit: Is this print needed?

File: packages/cloud_functions/lib/src/https_callable.dart [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        throw CloudFunctionsException._(code, message, details);
+      } else {
+        print('throwing generic exception');
```
Comment (bparrishMines) [inline packages/cloud_functions/lib/src/https_callable.dart]: nit: this one too

File: packages/cloud_functions/lib/src/https_callable.dart [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        throw Exception('Unable to call function ' + _functionName);
+      }
+    } catch (e) {
+      print(e);
```
Comment (bparrishMines) [inline packages/cloud_functions/lib/src/https_callable.dart]: nit: this one too

File: packages/cloud_functions/lib/src/https_callable.dart [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+      rethrow;
+    }
+  }
+
+  Duration _timeout;
+
+  Duration get timeout => _timeout;
+  set duration(Duration value) {
```
Comment (bparrishMines) [inline packages/cloud_functions/lib/src/https_callable.dart]: Should this be:

```suggestion
  set timeout(Duration value) {
```

File: packages/cloud_functions/lib/src/https_callable.dart (hunk 2 above)
Comment (collinjackson) [reply packages/cloud_functions/lib/src/https_callable.dart]: it's not, it's left over from previous implementation. removing it, thanks.

File: packages/cloud_functions/lib/src/https_callable.dart (hunk 3 above)
Comment (collinjackson) [reply packages/cloud_functions/lib/src/https_callable.dart]: it's not, it's left over from previous implementation. removing it, thanks.

File: packages/cloud_functions/lib/src/https_callable.dart (hunk 4 above)
Comment (collinjackson) [reply packages/cloud_functions/lib/src/https_callable.dart]: it's not, it's left over from previous implementation. removing it, thanks.

**Pull request merged**